src/javascript/app_2/pages/statement/statement.jsx

```diff
@@ -167,85 +108,17 @@ class Statement extends React.PureComponent {
         const {scrollTop, scrollHeight, clientHeight} = document.scrollingElement;
         const left_to_scroll = scrollHeight - (scrollTop + clientHeight);
 
-        if (left_to_scroll < 1000) {
-            this.loadNextChunk();
-        }
-    }
-
-    handleDateChange(e) {
-        if (e.target.value !== this.state[e.target.name]) {
-            this.reloadTable();
-        }
-        this.setState({
-            [e.target.name]: e.target.value,
-        });
-    }
-
-    loadNextChunk() {
-        const { chunk_size } = this.props;
-        const { chunks, data_source } = this.state;
-
-        if (data_source.length <= chunks * chunk_size) {
-            // all content is shown
-            return;
+        if (left_to_scroll < 2000) {
+            this.props.fetchNextBatch();
         }
-
-        this.setState({ chunks: chunks + 1 });
-
-        if (data_source.length <= (chunks + 1) * chunk_size) {
-            // last chunk has been loaded
-            this.fetchNextBatch();
-        }
-    }
-
-    fetchNextBatch() {
-        if (this.state.has_loaded_all || this.state.pending_request) return;
-
-        this.setState({ pending_request: true });
-
-        const currency     = Client.get('currency');
-
-        const { date_from, date_to } = this.state;
-
-        DAO.getStatement(
-            this.props.batch_size,
-            this.state.data_source.length,
-            {
-                ...date_from && {date_from: moment(date_from).unix()},
-                ...date_to   && {date_to: moment(date_to).add(1, 'd').subtract(1, 's').unix()},
-            }
-        ).then((response) => {
-            if (!this.el) return;
-
-            const formatted_transactions = response.statement.transactions
-                .map(transaction => getStatementData(transaction, currency));
-
-            this.setState({
-                data_source    : [...this.state.data_source, ...formatted_transactions],
-                has_loaded_all : formatted_transactions.length < this.props.batch_size,
-                pending_request: false,
-            });
-        });
-    }
-
-    reloadTable() {
-        this.setState(
-            {
-                data_source    : [],
-                has_loaded_all : false,
-                pending_request: false,
-                chunks         : 1,
-            },
-            this.fetchNextBatch
-        );
     }
 
     renderNoActivityMessage() {
         return (
             <div className='container'>
                 <div className='statement__no-activity-msg'>
                     {
-                        !this.state.date_from && !this.state.date_to
+                        !this.props.date_from && !this.props.date_to
```

Comment:
Code can be more readable when you define props explicitely.

1) You can define props 
`const { date_from, date_to } = this.props;`

2) use the props name directly
`date_from && date_to ? local blah blah`